.github/workflows/remote-copy.yml

```diff
@@ -157,20 +178,27 @@ jobs:
         id: copy
         env:
           REMOTE_RSYNC_FILE_LIST_PATH: ${{ vars.REMOTE_TMP_DIR }}/remote-copy-files-${{ github.run_id }}.log
+          LOCAL_RSYNC_FILE_LIST_PATH: ./files-copied.log
         # In this step, we rsync the files from the source to the target,
         # capturing the list of files copied as output of the step, since
         # it's used in the manifest step.
         run: |
           ssh ${{ secrets.SSH_USER }}@${{ secrets.SSH_HOST }} -i ${{ steps.ssh.outputs.private-key-path }} /bin/bash <<'EOT'
-          files=$(rsync --recursive --out-format="%n" \
+          rsync --recursive --out-format="${{ inputs.target }}/%n" \
               ${{ ! inputs.overwrite-target && '--ignore-existing' || '--update' }} \
               ${{ inputs.source }} ${{ inputs.target }} \
-            | tr '\n' ' ' \  # convert to space-separated for output
+            | grep --invert-match '/$' \
+            | tr '\n' ' ' \
             > ${{ env.REMOTE_RSYNC_FILE_LIST_PATH }}
-          )
           EOT
 
-          echo "paths=$(scp -q ${{ secrets.USER }}:${{ secrets.HOST }}:${{ env.REMOTE_RSYNC_FILE_LIST_PATH }} /dev/stdout) >> $GITHUB_OUTPUT
+          rsync -e 'ssh -i ${{ steps.ssh.outputs.private-key-path }}' \
+            ${{ secrets.SSH_USER }}@${{ secrets.SSH_HOST_DATA }}:${{ env.REMOTE_RSYNC_FILE_LIST_PATH }} \
+            ${{ env.LOCAL_RSYNC_FILE_LIST_PATH }}
+          cat ${{ env.LOCAL_RSYNC_FILE_LIST_PATH }}
+          paths=$(cat ${{ env.LOCAL_RSYNC_FILE_LIST_PATH }})
+          echo "paths are $paths"
```

Comment:
The paths are being `cat`ted to STDOUT twice.

Reply:
Done in https://github.com/ACCESS-NRI/model-config-inputs/pull/3/commits/c9e050456f71712f9e9d5b3ba9c007ce6871cc6f